ytfzf

```diff
@@ -326,6 +330,11 @@ download_thumbnails () {
 			url="$(echo $line | sed -E 's/^"//;s/\.png.*/\.png/')" 
 		else
 			name="$(echo "$line" | tr -d '"')"
+			if [ -e "$thumb_dir/$name.png" ]; then
+				i=$((i + 1))
+				continue
+			fi
+
 			{ curl -s "$url" > "$thumb_dir/$name.png" && [ $show_link_only -eq 0 ] && printf "."; } &
```

Comment:
How about just onle line
```sh
[ -e "$thumb_dir/$name.png" ] || { curl -s "$url" > "$thumb_dir/$name.png" && [ $show_link_only -eq 0 ] && printf "."; } &
```
Also with the current approach, i would go to i+1 and that would change odd to even and file names would mess up.